controller/machinedriver/machine_driver.go

```diff
@@ -95,5 +105,57 @@ func (m *Lifecycle) Remove(obj *v3.MachineDriver) (*v3.MachineDriver, error) {
 		}
 		logrus.Infof("Deleting schema %s done", schema.Name)
 	}
+	if err := m.createOrUpdateMachineForEmbeddedType(obj.Name + "config", false); err != nil {
+		return nil, err
+	}
 	return obj, nil
 }
+
+func (m *Lifecycle) createOrUpdateMachineForEmbeddedType(embbedType string, embedded bool) error {
+	machineSchema, err := m.schemaClient.Get("machineconfig", metav1.GetOptions{})
```

Comment:
You need to lock here because `Updated` will be called in parallel for different dynamicSchema IDs

Reply:
@ibuildthecloud now it is 🔒ed.